src/Features/Core/Portable/UseCollectionInitializer/ObjectCreationExpressionAnalyzer.cs

```diff
@@ -200,13 +200,30 @@ protected override void AddMatches(ArrayBuilder<TExpressionStatementSyntax> matc
             _syntaxFacts.GetPartsOfMemberAccessExpression(memberAccess, out var localInstance, out var memberName);
             _syntaxFacts.GetNameAndArityOfSimpleName(memberName, out var name, out var arity);
 
-            if (arity != 0 || !name.Equals(nameof(IList.Add)))
+            if (arity != 0 || !name.Equals(nameof(IList.Add)) || IsExplicitlyImplemented(memberName))
```

Comment:
❓ Will this run for cases where the local variable has the same type as the constructed object, e.g. when using `var`? If so, can we special case the behavior for this to avoid expensive calls to `GetSymbolInfo`?

Reply:
Good point, I'll check it.

Reply:
@sharwell Should we provide suggestion in this case:

```csharp
var obj = new ExpandoObject();
obj.Add("string", "v");
obj.Add("int", 1);
obj.Add("object", new { X=1, Y=2 });
```

This code won't compile but we have suggestion to use collection initialization

Reply:
I'm OK with either way in that scenario, so whatever's more efficient.